storage/src/journal/fixed.rs

```diff
@@ -229,6 +229,48 @@ impl<B: Blob, E: Storage<B>, const N: usize> Journal<B, E, N> {
         Ok(item_position + self.cfg.items_per_blob * newest_blob.0)
     }
 
+    /// Rewind the journal by `num` items.
+    ///
+    /// The journal is not synced after rewinding.
+    pub async fn rewind(&mut self, num: u64) -> Result<(), Error> {
+        let size = self.size().await?;
+        if size < num {
+            return Err(Error::InvalidRewind(size));
+        }
+        let rewind_to_size = size - num;
+        let rewind_to_blob_index = rewind_to_size / self.cfg.items_per_blob;
+        if rewind_to_blob_index < self.oldest_blob().0 {
+            return Err(Error::ItemPruned(rewind_to_size));
+        }
+
+        let chunk_size = size_of::<u32>() + N;
```

Comment:
We should switch to using `u32::SERIALIZED_LEN` (and anywhere else there is a size_of)

Reply:
As of Sunday-ish, I believe I removed all size_of usage everywhere

Reply:
done